src/ring/middleware/authorization.clj

```diff
@@ -0,0 +1,84 @@
+(ns ring.middleware.authorization
+  (:refer-clojure :exclude [update-keys])
+  (:require
+    [clojure.string :as str]
+    [ring.util.parsing :as parsing]))
+
+;; Copied from Clojure 1.11:
+(defn- update-keys
+  [m f]
+  (let [ret (persistent!
+              (reduce-kv (fn [acc k v] (assoc! acc (f k) v))
+                         (transient {})
+                         m))]
+    (with-meta ret (meta m))))
```

Comment:
Clojure is licensed under the EPL, I believe, and Ring-Headers under MIT. I also don't think we need this if we just lowercase the keys when we're parsing them.